CMakeLists.txt

```
@@ -48,34 +63,90 @@ endif()
 
 ## Add serial library
 add_library(${PROJECT_NAME} ${serial_SRCS})
+
 if(APPLE)
-    target_link_libraries(${PROJECT_NAME} ${FOUNDATION_LIBRARY} ${IOKIT_LIBRARY})
+    find_library(IOKIT_LIBRARY IOKit)
+    find_library(FOUNDATION_LIBRARY Foundation)
+
+	target_link_libraries(${PROJECT_NAME} ${FOUNDATION_LIBRARY} ${IOKIT_LIBRARY})
 elseif(UNIX)
-    target_link_libraries(${PROJECT_NAME} rt pthread)
+	target_link_libraries(${PROJECT_NAME} rt pthread)
 else()
-    target_link_libraries(${PROJECT_NAME} setupapi)
+	target_link_libraries(${PROJECT_NAME} setupapi)
 endif()
 
-## Uncomment for example
-add_executable(serial_example examples/serial_example.cc)
-add_dependencies(serial_example ${PROJECT_NAME})
-target_link_libraries(serial_example ${PROJECT_NAME})
-
 ## Include headers
-include_directories(include)
+target_include_directories(${PROJECT_NAME} PUBLIC $<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/include>)
```

```suggestion
target_include_directories(${PROJECT_NAME} PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>)
```

The use of `CMAKE_CURRENT_SOURCE_DIR`  is problematic when the project is used inside a bigger project via `add_subdirectory` , for  example when using the [`FetchContent`](https://cmake.org/cmake/help/v3.11/module/FetchContent.html) CMake module. Use of `CMAKE_CURRENT_SOURCE_DIR` should remove any ambiguity.